vowpalwabbit/core/src/loss_functions.cc

```diff
@@ -482,12 +483,18 @@ namespace VW
 std::unique_ptr<loss_function> get_loss_function(
     VW::workspace& all, const std::string& funcName, float function_parameter_0, float function_parameter_1)
 {
+  // For some loss functions, we want to know if all.set_minmax has been overwritten
+  // Check if the std::function object does not contain VW::details::noop_mm
+  using noop_mm_fn_ptr_type = typename std::add_pointer<decltype(VW::details::noop_mm)>::type;
+  auto* minmax_fptr = all.set_minmax.target<noop_mm_fn_ptr_type>();
+  bool set_minmax_is_not_noop = (minmax_fptr == nullptr || *minmax_fptr != VW::details::noop_mm);
```

Comment:
instead of this, couldn't you just do `set_minmax = nullptr` wherever we set it to noop_mm? Does that break some other logic elsewhere?

Reply:
`noop_mm` isn't a new function here - it was there before and I just changed the signature. I think the purpose of having it was to avoid needing to do a nullptr check every time we want to call the set_minmax function.

The alternative way of just doing the nullptr check should also be fine. I'll try it out and see if it'll let me get rid of the ugly std::function target() call, which requires figuring out a specific "pointer to your function pointer" type.

Reply:
`noop_mm` isn't a new function here - it was there before and I just changed the signature. I think the purpose of having it was to avoid needing to do a nullptr check every time we want to call the set_minmax function.

The alternative way of just doing the nullptr check should also be fine. I'll try it out and see if it'll let me get rid of the ugly std::function target() call, which requires figuring out a specific "pointer to your function pointer" type.